cameo/flux_analysis/analysis.py

```diff
@@ -226,24 +226,31 @@ def _flux_variability_analysis(model, reactions=None):
     else:
         reactions = model._ids_to_reactions(reactions)
     fva_sol = OrderedDict()
-    [lb_flag, ub_flag] = [False, False]
+    lb_flags = dict()
     with TimeMachine() as tm:
         model.change_objective(S.Zero, time_machine=tm)
         for reaction in reactions:
+            lb_flags[reaction.id] = False
             fva_sol[reaction.id] = dict()
-            model.solver.objective.set_linear_coefficients({reaction.forward_variable: 1., reaction.reverse_variable: -1.})
+            model.solver.objective.set_linear_coefficients({reaction.forward_variable: 1.,
+                                                            reaction.reverse_variable: -1.})
             model.objective.direction = 'min'
             try:
                 solution = model.solve()
                 fva_sol[reaction.id]['lower_bound'] = solution.f
             except Unbounded:
                 fva_sol[reaction.id]['lower_bound'] = -numpy.inf
             except Infeasible:
-                lb_flag = True
-            model.solver.objective.set_linear_coefficients({reaction.forward_variable: 0., reaction.reverse_variable: 0.})
+                lb_flags[reaction.id] = True
+            model.solver.objective.set_linear_coefficients({reaction.forward_variable: 0.,
+                                                            reaction.reverse_variable: 0.})
+
+            assert model.objective.expression == 0, model.objective.expression
 
         for reaction in reactions:
-            model.solver.objective.set_linear_coefficients({reaction.forward_variable: 1., reaction.reverse_variable: -1.})
+            ub_flag = False
+            model.solver.objective.set_linear_coefficients({reaction.forward_variable: 1.,
+                                                            reaction.reverse_variable: -1.})
```

Comment:
`model.objective.direction = 'max'` could be moved outside the loop for a small performance boost